schemas/component/component.schema

```diff
@@ -9,6 +9,48 @@
         },
         {
             "properties": {
+                "$policies": {
+                    "type": "array",
+                    "items": { 
+                        "oneOf": [
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "type": {
+                                        "type": "string", 
+                                        "string": "requiresAction"
+                                    },
+                                    "kinds": {
+                                        "type":"array",
+                                        "items": {
+                                            "type": "string"
+                                        }
+                                    }
+                                },
+                                "required": [
+                                    "type",
+                                    "kinds"
+                                ]
+                            }, 
+                            {
+                                "type": "object",
+                                "properties": {
+                                    "type": {
+                                        "type": "string", 
+                                        "enum": [
+                                            "lastAction",
+                                            "interactive",
+                                            "notInteractive"
+                                        ]
+                                    }
+                                },
+                                "required": [
+                                    "type"
+                                ]
+                            }
+                        ]
```

Comment:
Why did you make this an object and not just an enum?  Do you anticipate needing to add other properties to these policies?

Reply:
@chrimc62 This just does not feel right to me:

```
"$policies": [
	{
		"type": "mustInclude",
		"kinds": [
			"Microsoft.xyz"
		]
	},
	"lastAction"
]
```

Consumption of a json array of objects guaranteed to have 'type', is less complex.  Do you think this is a mistake?

Reply:
I'm fine either way.  The advantage of an enum is less typing and immediate discovery in the list for customers writing schemas.  The disadvantage is mainly on the consumption side where you have to deal with strings and objects.  The only consumers are likely to be us so I tend to favor making it easier for customers writing schemas even if we have to deal with both, but either solution is fine.  

---
In reply to: [579387313](https://github.com/microsoft/botframework-sdk/pull/6208#discussion_r579387313) [](ancestors = 579387313)

Reply:
Part of the reason I don't think just having objects is easier to consume is because you have to deal with different shape objects.  Strings are just a simple version of that and have the advantage of simpler to discover and type.

---
In reply to: [579391141](https://github.com/microsoft/botframework-sdk/pull/6208#discussion_r579391141) [](ancestors = 579391141,579387313)

Reply:
Change to object

Thanks for this feedback @chrimc62